<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>rare_variants</title>
  <meta name="description" content="Posting ground for my projects and musings within grad school (and hopefully beyond!)">

  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.2/css/all.css" integrity="sha384-oS3vJWv+0UjzBfQzYUhtDYW+Pj2yciDJxpsK1OYPAYjqT085Qq/1cq5FLXAZQ7Ay" crossorigin="anonymous">
  <link rel="stylesheet" href="/assets/css/main.css">
</head>

  <script type="text/javascript" async
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML">
  </script>

  <body>
      <section id="page_header">
        <div class="shortParticle">
<div id="particles-js">
  <div class="page-header">
      <h1>
        <span class="site-title">Scott Kulm</span>
        <span class="site-description">Graduate Student</span>
	<!-- span class="site-back-home"><a href="https://kulmsc.github.io">Home</a></span> -->
	 <a class="site-back-home" href="https://kulmsc.github.io">Home</a>
      </h1>
  </div>
</div>
</div>

      </section>
    <div class="forMarkdown">
      <h1 id="discovery-and-association-of-pathogenic-cardiac-rare-variants">Discovery and Association of Pathogenic Cardiac Rare Variants</h1>

<h2 id="introduction">Introduction</h2>

<p>The primary tool for analyzing genetic data is a linear regression.  While the regression may be in many different forms, fundamentally what is happening is a comparison of a genetic-type vector against some outcome or phenotype.  This process is flexible and powerful, however begins to break down when the genetic-type vector is skewed to one class.  Or in other words, the vast majority of individuals have one type of genetic identity (at that variant) while a small group of individuals have another type of genetic identity. While there is a good mathematical explanation to back this up, the underlying reason we have difficulty analyzing this skewed dataset is that it represents a small sample size.  The few people with the rare variant could have a disease because of the variant, or perhaps their disease status is just random chance - with a small sample size it is difficult to determine which is true.</p>

<p>A single rare variant can impart a relatively huge impact on a relativey tiny number of individuals.  Therefore, we cannot give up on identifying whether a rare variant does impact or lead to disease just because regression does not work.  Rather we need to develop other tools.  In this research pursuit two general types of identification methods have been developed.  The first simply looks at case reports from clinical settings.  This slightly old fashioned approach can be though of as a collection of anecdotes for a given variant which, when viewed together, can be used to suggest that a variant does have some true, causal relationship to a disease.  For example, if ten doctors and ten different hospitals all stumble upon ten patients who have an odd cardiac condition and the same super rare genetic variant then we can conclude that super rare genetic variant is likely doing something important.  These reports or anecdotes are collected within a resource called <a href="https://www.ncbi.nlm.nih.gov/clinvar/">ClinVar</a>.  The second approach employs sophisticated algorithms which try to predict how the rare variant affects downstream biology.  Namely, does the rare variant cause a proten to be misfolded and disfunctional.  Many different algorithms exist, each of which report a different severity or pathogenicity score.</p>

<p>With many methods available to determine whether a rare variant is pathogenic, it may seem impossible to come up with a single, clear methedology.  Fortunately, many very bright individuals have developed just such a methedology.  The American College of Medical Genetics (ACMG) provide a <a href="https://www.acmg.net/docs/Standards_Guidelines_for_the_Interpretation_of_Sequence_Variants.pdf">flowchart with detailed accompanying descriptions</a> that help one classify a variant as being either pathogenic or non-pathogenic.  The only problem</p>

<p><img src="/assets/img/acmg_flow.png" alt="ACMG definition pathogenic" /></p>

<p>Unfortunately for my purposes, this flow chart requires a great deal of labor (multiple hours) just to classify a single variant as pathogenic.  While resources are being developed that alleviate this load, for example <a href="https://clinicalgenome.org/">ClinGen</a>, they are not yet at the point to help me out.  Therefore, for this project I defined a variant as being pathogenic if it met all three conditions (then I went on to see if these pathogenic variants associate to adverse cardiac conditions).</p>

<h2 id="methods">Methods</h2>

<p>To begin, I defined pathogenicity from the three following conditions.</p>

<ol>
  <li>High Confidence Pathogenic According to LOFTEE - Ensembl has released a piece of software entitled <a href="https://useast.ensembl.org/info/docs/tools/vep/index.html">Variant Effect Predictor</a> (VEP), which does exactly what it sounds like.  Although, VEP does not report a simple binary rating of patogenicity, which is likely why a recent add-on was created called <a href="https://github.com/konradjk/loftee">LOFTEE</a> which attempts to report whether a variant can be classified with high confidence a loss of function variant.</li>
  <li>Listed as Pathogenic or Likely Pathogenic Within ClinVar -</li>
  <li>Located on a Gene That Invitae Uses in Their Cardiac Panel - Invitae is a private, well-respected company that creates clinical-grade reports on a patient’s genetics.  We figured that they have a good sense of what genes potentially influence adverse cardiac conditions.  The genes we specifically used came from the <a href="https://www.invitae.com/en/providers/test-catalog/test-02101">Invitae Arrhythmia and Cardiomyopathy Comprehensive Panel</a>.</li>
</ol>


    </div>

      </div>
      <footer class="footer">
    <p>&copy; Scott Kulm</p>
    <p>Build with Jekyll and <span class="love">❤</span> by <a href="https://github.com/nrandecker">Nathan Randecker</a></p>
</footer>
<script src="//cdn.jsdelivr.net/particles.js/2.0.0/particles.min.js"></script>
<script src="/assets/js/sweet-scroll.min.js"></script>
<script src="/assets/js/main.js"></script>
<!-- Google Analytics -->

<script>
var dnt = navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack;
if (dnt != "1" && dnt != "yes") {
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
ga('create', '', 'auto');
ga('send', 'pageview');
}
</script>



  </body>
