<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>qgg_notes</title>
  <meta name="description" content="Posting ground for my projects and musings within grad school (and hopefully beyond!)">

  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.2/css/all.css" integrity="sha384-oS3vJWv+0UjzBfQzYUhtDYW+Pj2yciDJxpsK1OYPAYjqT085Qq/1cq5FLXAZQ7Ay" crossorigin="anonymous">
  <link rel="stylesheet" href="/assets/css/main.css">
</head>

  <script type="text/javascript" async
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML">
  </script>

  <body>
      <section id="page_header">
        <div class="shortParticle">
<div id="particles-js">
  <div class="page-header">
      <h1>
        <span class="site-title">Scott Kulm</span>
        <span class="site-description">Graduate Student</span>
      </h1>
  </div>
</div>
</div>

      </section>
    <div class="forMarkdown">
      <h1 id="qgg-labs">qgg-labs</h1>

<p>These are all the computer labs I wrote for a class I TA’d entitled, Quantiative Genetics and Genomics at Cornell University and Weill Cornell Medicine.  The base of the computer labs were written by Zoe Zhao, you can get see these labs at <a href="https://github.com/zoezhao997/BTRY6830">her github</a>.  There are also video lectures that well correspond to the fundamental mathematics and biology of each lab.  You can find these videos at <a href="http://mezeylab.cb.bscb.cornell.edu/Classes.aspx">Mezey Lab</a>.</p>

<p>There may still be some bugs within the labs, but for the most part they are great.  From basic R to MCMC, these labs provide a way to learn R in the context of genetics.  The labs are all based in R-Markdown, which is something I have not seen in many other R lessons so hopefully you can take advantage of that fact here.  All of these labs are on my <a href="https://github.com/kulmsc/qgg-labs">github</a>.  The contents of the labs are as follows:</p>

<h3 id="lab-1---r-basics"><a href="https://github.com/kulmsc/qgg-labs/tree/master/Lab1">Lab 1</a> - R Basics</h3>
<p>R as a calculator, Matrices and data frames</p>

<h3 id="lab-2---r-slightly-beyond-basics"><a href="https://github.com/kulmsc/qgg-labs/tree/master/Lab2">Lab 2</a> - R Slightly Beyond Basics</h3>
<p>R Markdown, Functions, For Loops, If Statements, Vector and Matrix calculations</p>

<h3 id="lab-3---r-more-beyond-basics"><a href="https://github.com/kulmsc/qgg-labs/tree/master/Lab3">Lab 3</a> - R More Beyond Basics</h3>
<p>Plotting, Probability Distributions</p>

<h3 id="lab-4---r-further-beyond-basics"><a href="https://github.com/kulmsc/qgg-labs/tree/master/Lab4">Lab 4</a> - R Further Beyond Basics</h3>
<p>Boolean Data, Fancy vector indexing, Dealing with missing data</p>

<h3 id="lab-5---r-beyond-basics"><a href="https://github.com/kulmsc/qgg-labs/tree/master/Lab5">Lab 5</a> - R Beyond Basics</h3>
<p>Pseudo-random numbers, Pasting, Speeding up your code</p>

<h3 id="lab-6---genetics"><a href="https://github.com/kulmsc/qgg-labs/tree/master/Lab6">Lab 6</a> - Genetics!</h3>
<p>Reading in genotype data, Coding genotype data</p>

<h3 id="lab-7---genetic-association-testing"><a href="https://github.com/kulmsc/qgg-labs/tree/master/Lab7">Lab 7</a> - Genetic Association Testing</h3>
<p>Performing hypothesis test on variants, Producing Manhattan plots</p>

<h3 id="lab-8---gwas"><a href="https://github.com/kulmsc/qgg-labs/tree/master/Lab8">Lab 8</a> - GWAS</h3>
<p>Performing many association tests, QQ-Plots, Multiple hypothesis testing corrections, Principal Components Analysis</p>

<h3 id="lab-9---real-eqtl-analysis"><a href="https://github.com/kulmsc/qgg-labs/tree/master/Lab9">Lab 9</a> - Real eQTL Analysis</h3>
<p>Regrssion within R, Adding covariates, Converting data for PLINK</p>

<h3 id="lab-10---advanced-gwas"><a href="https://github.com/kulmsc/qgg-labs/tree/master/Lab10">Lab 10</a> - Advanced GWAS</h3>
<p>Adding covariates rigerously, Logistic Regression</p>

<h3 id="lab-11---irls-algorithm"><a href="https://github.com/kulmsc/qgg-labs/tree/master/Lab11">Lab 11</a> - IRLS Algorithm</h3>
<p>Implementing the IRLS Algorithm, Making algorithms efficient</p>

<h3 id="lab-12---em-algorithm"><a href="https://github.com/kulmsc/qgg-labs/tree/master/Lab12">Lab 12</a> - EM Algorithm</h3>
<p>Linear mixed models, Implementing EM Algorithm, Timing your code</p>

<h3 id="lab-13---mcmc-algorithm"><a href="https://github.com/kulmsc/qgg-labs/tree/master/Lab13">Lab 13</a> - MCMC Algorithm</h3>
<p>Explanation of Markov Chains and Monte Carlo Sampling, Implementing MCMC Algorithm</p>

    </div>

      </div>
      <footer class="footer">
    <p>&copy; Scott Kulm</p>
    <p>Build with Jekyll and <span class="love">❤</span> by <a href="https://github.com/nrandecker">Nathan Randecker</a></p>
</footer>
<script src="//cdn.jsdelivr.net/particles.js/2.0.0/particles.min.js"></script>
<script src="/assets/js/sweet-scroll.min.js"></script>
<script src="/assets/js/main.js"></script>
<!-- Google Analytics -->

<script>
var dnt = navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack;
if (dnt != "1" && dnt != "yes") {
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
ga('create', '', 'auto');
ga('send', 'pageview');
}
</script>



  </body>
