<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>qgg_notes</title>
  <meta name="description" content="Posting ground for my projects and musings within grad school (and hopefully beyond!)">

  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.2/css/all.css" integrity="sha384-oS3vJWv+0UjzBfQzYUhtDYW+Pj2yciDJxpsK1OYPAYjqT085Qq/1cq5FLXAZQ7Ay" crossorigin="anonymous">
  <link rel="stylesheet" href="/assets/css/main.css">
</head>

  <script type="text/javascript" async
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML">
  </script>

  <body>
      <section id="page_header">
        <div class="shortParticle">
<div id="particles-js">
  <div class="page-header">
      <h1>
        <span class="site-title">Scott Kulm</span>
        <span class="site-description">Graduate Student</span>
      </h1>
  </div>
</div>
</div>

      </section>
    <div class="forMarkdown">
      <h1 id="qgg-labs">qgg-labs</h1>

<p>These are all the computer labs I wrote for a class I TA’d entitled, Quantiative Genetics and Genomics at Cornell University and Weill Cornell Medicine.  The base of the computer labs were written by Zoe Zhao, you can get see these labs at <a href="https://github.com/zoezhao997/BTRY6830">her github</a>.  There are also video lectures that well correspond to the fundamental mathematics and biology of each lab.  You can find these videos at <a href="http://mezeylab.cb.bscb.cornell.edu/Classes.aspx">Mezey Lab</a>.</p>

<p><strong>There</strong> may still be some bugs within the labs, but for the most part they are great.  From basic R to MCMC, these labs provide a way to learn R in the context of genetics.  The labs are all based in R-Markdown, which is something I have not seen in many other R lessons so hopefully you can take advantage of that fact here.  <strong>All of these labs are on my <a href="https://github.com/kulmsc/qgg-labs">github</a></strong>.  The contents of the labs are as follows:</p>

<h3 id="lab-1---r-basics">Lab 1 - R Basics</h3>

<ul>
  <li>R as a calculator</li>
  <li>Matrices and data frames</li>
</ul>

<h3 id="lab-2---r-slightly-beyond-basics">Lab 2 - R Slightly Beyond Basics</h3>

<ul>
  <li>R Markdown</li>
  <li>Functions</li>
  <li>For Loops</li>
  <li>If Statements</li>
  <li>Vector and Matrix calculations</li>
</ul>

<h3 id="lab-3---r-more-beyond-basics">Lab 3 - R More Beyond Basics</h3>
<ul>
  <li>Plotting</li>
  <li>Probability Distributions</li>
</ul>

<h3 id="lab-4---r-further-beyond-basics">Lab 4 - R Further Beyond Basics</h3>
<ul>
  <li>Boolean Data</li>
  <li>Fancy vector indexing</li>
  <li>Dealing with missing data</li>
</ul>

<h3 id="lab-5---r-beyond-basics">Lab 5 - R Beyond Basics</h3>
<ul>
  <li>Pseudo-random numbers</li>
  <li>Pasting</li>
  <li>Speeding up your code</li>
</ul>

<h3 id="lab-6---genetics">Lab 6 - Genetics!</h3>
<ul>
  <li>Reading in genotype data</li>
  <li>Coding genotype data</li>
</ul>

<h3 id="lab-7---genetic-association-testing">Lab 7 - Genetic Association Testing</h3>
<ul>
  <li>Performing hypothesis test on variants</li>
  <li>Producing Manhattan plots</li>
</ul>

<h3 id="lab-8---gwas">Lab 8 - GWAS</h3>
<ul>
  <li>Performing many association tests</li>
  <li>QQ-Plots</li>
  <li>Multiple hypothesis testing corrections</li>
  <li>Principal Components Analysis</li>
</ul>

<h3 id="lab-9---real-eqtl-analysis">Lab 9 - Real eQTL Analysis</h3>
<ul>
  <li>Regrssion within R</li>
  <li>Adding covariates</li>
  <li>Converting data for PLINK</li>
</ul>

<h3 id="lab-10---advanced-gwas">Lab 10 - Advanced GWAS</h3>
<ul>
  <li>Adding covariates rigerously</li>
  <li>Logistic Regression</li>
</ul>

<h3 id="lab-11---irls-algorithm">Lab 11 - IRLS Algorithm</h3>
<ul>
  <li>Implementing the IRLS Algorithm</li>
  <li>Making algorithms efficient</li>
</ul>

<h3 id="lab-12---em-algorithm">Lab 12 - EM Algorithm</h3>
<ul>
  <li>Linear mixed models</li>
  <li>Implementing EM Algorithm</li>
  <li>Timing your code</li>
</ul>

<h3 id="lab-13---mcmc-algorithm">Lab 13 - MCMC Algorithm</h3>
<ul>
  <li>Explanation of Markov Chains and Monte Carlo Sampling</li>
  <li>Implementing MCMC Algorithm</li>
</ul>


    </div>

      </div>
      <footer class="footer">
    <p>&copy; Scott Kulm</p>
    <p>Build with Jekyll and <span class="love">❤</span> by <a href="https://github.com/nrandecker">Nathan Randecker</a></p>
</footer>
<script src="//cdn.jsdelivr.net/particles.js/2.0.0/particles.min.js"></script>
<script src="/assets/js/sweet-scroll.min.js"></script>
<script src="/assets/js/main.js"></script>
<!-- Google Analytics -->

<script>
var dnt = navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack;
if (dnt != "1" && dnt != "yes") {
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
ga('create', '', 'auto');
ga('send', 'pageview');
}
</script>



  </body>
